Fix: Add space between async and param on fix (fixes #8682)

soda0289 opened the pull request:
**What is the purpose of this pull request? (put an "X" next to item)**
[ X ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (http://eslint.org/docs/developer-guide/contributing/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

**What changes did you make? (Give an overview)**
Update the fixer code to add space between async and the arrow function parameter when they are side by side.

**Is there anything you'd like reviewers to focus on?**
Nothing in particular

not-an-aardvark commented on lib/rules/arrow-parens.js:
```diff
@@ -51,13 +51,32 @@ module.exports = {
         function parens(node) {
-            const token = sourceCode.getFirstToken(node, node.async ? 1 : 0);
+            const isAsync = node.async;
+            const token = sourceCode.getFirstToken(node, isAsync ? 1 : 0);
```
I know this was here before, but can you give this variable a more descriptive name? It's not clear which token it's referring to.

not-an-aardvark commented on lib/rules/arrow-parens.js:
```diff
@@ -51,13 +51,32 @@ module.exports = {
+            function fixParamsWithParenthesis(fixer) {
+                const paramToken = context.getTokenAfter(token);
+                const closingParenToken = context.getTokenAfter(paramToken);
+                const asyncToken = isAsync ? context.getTokenBefore(token) : null;
```
Can you use `sourceCode.getTokenBefore` and `sourceCode.getTokenAfter` instead of `context.getTokenBefore` and `context.getTokenAfter`? The latter two are deprecated.

soda0289 commented:
Changes:
- changed the `token` variable to `firstTokenOfParam`, which might be a opening paren or identifier.
- used source.getToken functions instead of context.getToken
- Removed extra blank lines

kaicataldo commented:
Thanks for contributing!
